crates/sui-json-rpc/src/api/mod.rs

```diff
@@ -60,10 +60,19 @@ pub fn cap_page_limit(limit: Option<usize>) -> usize {
     }
 }
 
-pub fn validate_limit(limit: Option<usize>, max: usize) -> Result<usize, anyhow::Error> {
+#[derive(Debug, Error)]
+pub enum InvalidLimitReasons {
+    #[error("`exceeds limit of `{0}`")]
+    Exceeded(usize),
+
+    #[error("must be at least `{0}`")]
+    TooSmall(usize),
+}
+
+pub fn validate_limit(limit: Option<usize>, max: usize) -> Result<usize, InvalidLimitReasons> {
     match limit {
-        Some(l) if l > max => Err(anyhow!("Page size limit {l} exceeds max limit {max}")),
-        Some(0) => Err(anyhow!("Page size limit cannot be smaller than 1")),
+        Some(l) if l > max => Err(InvalidLimitReasons::Exceeded(max)),
```

Comment:
the previous string included more specific details that helps users understand the error